### ISSUE TITLE
SLE

### ISSUE THREAD

**Issue by RX28666** [skepsun/SAGN_with_SLE#4]
Hello,

Thanks for your interesting work! I have a question about SLE, after getting new pseudo labels each time, do I need to reset the parameters and retrain the model from start or just need to keep training on the current model which already trained before pseudo labels added? Thanks

**Comment by skepsun** [comment]
Hi, @RX28666 
In SLE, after generating new pseudo labels in each stage, you need to train a new model (with exactly the same hyperparameters) from scratch. 
In addition, you can also try to keep training on the current trained model.

**Comment by RX28666** [comment]
Hello,

Thanks for your reply. Is training a new model always better than keep training the current model? I am wondering about the reason for this. Thanks again.

**Comment by skepsun** [comment]
We have not fully compared two conditions. We prefer to train a new model because the current model  has been trained with full training data and has the same knowledge of generated pseudo labels.  We suppose a clean new model can better capture the potential information in the augmented training set. However, anyway, it is nice if you can try to keep training the current model since only experiments can confirm which is better.

**Comment by RX28666** [comment]
Appreciate your help. It makes sense. I will try both to check results.
